src/Moq/LookupOrFallbackDefaultValueProvider.cs

```diff
@@ -32,26 +32,26 @@ namespace Moq
 	[EditorBrowsable(EditorBrowsableState.Advanced)]
 	public abstract class LookupOrFallbackDefaultValueProvider : DefaultValueProvider
 	{
-		private Dictionary<Type, Func<Type, Mock, object>> factories;
+		private Dictionary<object, Func<Type, Mock, object>> factories;
 
 		/// <summary>
 		/// Initializes a new instance of the <see cref="LookupOrFallbackDefaultValueProvider"/> class.
 		/// </summary>
 		protected LookupOrFallbackDefaultValueProvider()
 		{
-			this.factories = new Dictionary<Type, Func<Type, Mock, object>>()
+			this.factories = new Dictionary<object, Func<Type, Mock, object>>()
 			{
 				[typeof(Task)] = CreateTask,
 				[typeof(Task<>)] = CreateTaskOf,
 				[typeof(ValueTask<>)] = CreateValueTaskOf,
-				[typeof(ValueTuple<>)] = CreateValueTupleOf,
-				[typeof(ValueTuple<,>)] = CreateValueTupleOf,
-				[typeof(ValueTuple<,,>)] = CreateValueTupleOf,
-				[typeof(ValueTuple<,,,>)] = CreateValueTupleOf,
-				[typeof(ValueTuple<,,,,>)] = CreateValueTupleOf,
-				[typeof(ValueTuple<,,,,,>)] = CreateValueTupleOf,
-				[typeof(ValueTuple<,,,,,,>)] = CreateValueTupleOf,
-				[typeof(ValueTuple<,,,,,,,>)] = CreateValueTupleOf,
+				["System.ValueTuple`1"] = CreateValueTupleOf,
+				["System.ValueTuple`2"] = CreateValueTupleOf,
+				["System.ValueTuple`3"] = CreateValueTupleOf,
+				["System.ValueTuple`4"] = CreateValueTupleOf,
+				["System.ValueTuple`5"] = CreateValueTupleOf,
+				["System.ValueTuple`6"] = CreateValueTupleOf,
+				["System.ValueTuple`7"] = CreateValueTupleOf,
+				["System.ValueTuple`8"] = CreateValueTupleOf,
```

Comment:
What this means is that Moq now recognises `System.ValueTuple<…>` types by name instead of by strict type identity. This mimicks what the C# compiler already does.

(Unlike the C# compiler, Moq however won't validate the structural makeup of the matched types, so Moq's default value providers will not work correctly with such user types that don't have the expected constructor signature.)